Scripts/Characters/Player/Player.cs

```diff
@@ -117,17 +118,40 @@ private void HandleDash()
             {
                 _animatedSprite.Stop();
                 _animatedSprite.Play("jump");
-                Position -= dashDistance;
+                StartDash(-dashDistance);
             }
             else if (_lastHorizontalInput == "right")
             {
                 _animatedSprite.Stop();
                 _animatedSprite.Play("jump");
-                Position += dashDistance;
+                StartDash(dashDistance);
             }
         }
     }
 
+    private async void StartDash(Vector2 dashDistance)
+    {
+        float elapsedTime = 0;
+        Vector2 dashVelocity = dashDistance / _dashTime;
+
+        while (elapsedTime < _dashTime)
+        {
+            Vector2 motion = dashVelocity * (float)GetProcessDeltaTime();
+            KinematicCollision2D collision = MoveAndCollide(motion);
+
+            if (collision != null)
+            {
+                // Stop dashing if a collision occurs
+                break;
+            }
+
+            elapsedTime += (float)GetProcessDeltaTime();
+            await ToSignal(GetTree(), "process_frame");
+        }
+
+        _isDashing = false;
+    }
+
     private void HandleGravity(double delta)
```

Comment:
Ensure gravity is applied only when the player is not on the floor. Consider resetting vertical velocity when the player is on the floor.

Reply:
Don't do this lmao, tried it, gets you stuck right to the floor, we need to set it when the player collides with the bottom of their collision rectangle instead, and only on the first collision, so something like `if (velocity.Y != 0 && IsOnGround()) velocity.Y = 0;  This may not even matter though, I think for now this is out of scope, so I'm closing this comment after we talk about it.